Source/WebKit/GPUProcess/graphics/RemoteGraphicsContextGLFunctionsGenerated.h

```diff
@@ -379,21 +379,21 @@
         assertIsCurrent(workQueue());
         Vector<GCGLfloat, 16> value(valueSize, 0);
         protectedContext()->getFloatv(pname, value);
-        completionHandler(std::span<const float>(reinterpret_cast<const float*>(value.data()), value.size()));
+        completionHandler(spanReinterpretCast<const float>(value.span()));
     }
     void getIntegerv(uint32_t pname, size_t valueSize, CompletionHandler<void(std::span<const int32_t>)>&& completionHandler)
     {
         assertIsCurrent(workQueue());
         Vector<GCGLint, 4> value(valueSize, 0);
         protectedContext()->getIntegerv(pname, value);
-        completionHandler(std::span<const int32_t>(reinterpret_cast<const int32_t*>(value.data()), value.size()));
+        completionHandler(spanReinterpretCast<const int32_t>(value.span()));
     }
     void getIntegeri_v(uint32_t pname, uint32_t index, CompletionHandler<void(std::span<const int32_t, 4>)>&& completionHandler) // NOLINT
     {
         assertIsCurrent(workQueue());
         std::array<GCGLint, 4> value { };
         protectedContext()->getIntegeri_v(pname, index, value);
-        completionHandler(std::span<const int32_t, 4>(reinterpret_cast<const int32_t*>(value.data()), value.size()));
+        completionHandler(spanReinterpretCast<const int32_t>(std::span(value)));
```

Comment:
I think we could have also used `std::span<const GCGLink>(value)` here and avoided the changes to spanReinterpretCast() right?

Reply:
I'm not super familiar with this code; I was just trying to mimic the existing design of the autogenerator.

GCGLint is typedef'd to int, so you can avoid the cast on any platform where int == int32_t, which is probably most platforms. The same appears to be true of GCGLfloat and GCGLuint, at least.

Should I just remove all the casts? fear removing them, since I don't know why they were there to begin with. Maybe I'll break a platform I don't know about.

Reply:
I didn't suggest removing any cases. My concern is that you're constructing a non-const span only to cast it to a const span. In order to achieve that, you had to modify `spanReinterpretCast`. However, it seems you could simply construct a const span right away and `spanReinterpretCast` would only cast between const spans:
```
completionHandler(spanReinterpretCast<const int32_t>(std::span<const GCGLink>(value)));
```

Reply:
I see. I think the autogenerator has enough metadata to do that. I can give it a shot.

Is it bad to cast *to* const?

Reply:
> Is it bad to cast to const?

Not really. Just seemed unfortunate to create a non-const span only to cast it to const right away.